text/0054-kerberos-buildpack.md

```diff
@@ -0,0 +1,55 @@
+# Introduce Kerberos Buildpack
+
+## Summary
+
+Introduce a buildpack that installs the [MIT
+Kerberos](https://kerberos.org/dist/) libraries and executables such that
+applications authenticating with Kerberos can do so on any of the supported
+stacks.
+
+## Motivation
+
+Kerberos is currently supported for applications built on top of the Full
+stack. This is enabled through the addition of the `krb5-user` package in our
+stack definition. Unfortunately, the Full stack also includes a number of other
+packages that many application developers would like to avoid installing.
+
+Creating a new Kerberos buildpack will allow these developers to make a more
+focused addition to their application image be ensuring that only those
+packages relating to Kerberos are installed. It will also allow the project to
+support Kerberos on top of stacks that are not the Full stack without needing
+to also include that package in these other stack definitions.
+
+## Detailed Explanation
+
+A new repository will be created at `paketo-buildpacks/kerberos`. This repo
+will house a buildpack that has the singular purpose of installing the Kerberos
+libraries and executables into a layer made available to other buildpacks or at
+runtime. The buildpack will include Kerberos as a precompiled package that can
+be dropped into a layer with support for Jammy only. The buildpack will follow
+the existing provide/requires conventions, exposing a dependency called
+`kerberos`.
+
+## Rationale and Alternatives
+
+An alternative would be to add the `krb5-user` package to the Base stack
+images. Avoiding this would be preferrable as we should try not to increase the
+scope or vectors of attack for our most commonly used stack image.
+
+## Implementation
+
+The implementation will use all of the new dependency management infrastructure
+and so be contained to this one repository. The dependency will need to be
+compiled against a Jammy stack and hosted by ourselves as no precompiled
+distribution has been found.
+
+## Prior Art
+
+The most immediately similar example of this kind of library-installing
+buildpack is the ICU buildpack that is used within the .NET Core language
+family.
+
+## Unresolved Questions and Bikeshedding
+
+* Do we need to backport support to Bionic?
```

Comment:
I think it would be worth supporting Bionic for the next few months if its reasonably easy to compile the dependency against both stacks and get support. Are you aware of if different libraries will be needed for Bionic as well?

Reply:
I don't believe there are any major differences in compiling this for Bionic vs Jammy. Just that its a bunch of C code and so likely needs to be compiled independently on each version.

Reply:
I'd probably start with Jammy, and depending on how long it takes to release and how long we have left before Bionic goes out of support (i.e. April 2023), we could evaluate the benefit of shipping a bionic dependency too.

Reply:
Worth noting that users relying on .NET 3.1 apps can't use jammy, because the runtime itself isn't compatible. So by not supporting bionic, we're saying that Kerberos and .NET 3.1 support are mutually exclusive features. This isn't terrible, since [.NET 3.1 technically EOLs December 13, 2022](https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core). But I wouldn't be surprised if there are still .NET 3.1 users in the wild long after its EOL date.

Reply:
But wouldn't we remove 3.1 after that date anyway? I feel like the overlap of a Kerberos buildpack existing and us still delivering support for .NET 3.1 would be infinitesimally small if non-existent, no?

Reply:
I think that's true. Just want to be clear about expectations-setting here for users.

Reply:
I think we should ignore Bionic until we get feedback that users want it. I think that by the time we get a buildpack out, it will be close to the EOL date for Bionic anyway and any users trying this out will already be looking to run on Jammy. I'll update the RFC to state that we are explicitly not going to support Bionic in the absence of user demand.